### PR TITLE
br: support backup txn kv data 

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/38506

Problem Summary:
support backup and restore txn kv.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/backup_txn.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+		"backup sst file compression algorithm, value can be one of 'lz4|zstd|snappy'")
+	command.Flags().Bool(flagRemoveSchedulers, false,
+		"disable the balance, shuffle and region-merge schedulers in PD to speed up backup")
+	// This flag can impact the online cluster, so hide it in case of abuse.
+	_ = command.Flags().MarkHidden(flagRemoveSchedulers)
+}
+
+// ParseFromFlags parses the txn kv backup&restore common flags from the flag set.
+func (cfg *TxnKvConfig) ParseFromFlags(flags *pflag.FlagSet) error {
+	if len(cfg.StartKey) > 0 && len(cfg.EndKey) > 0 && bytes.Compare(cfg.StartKey, cfg.EndKey) >= 0 {
```
Comment (Leavrth) [inline br/pkg/task/backup_txn.go:57]: why delete `flags.GetString(flagStartKey)`? 😕

File: br/pkg/task/restore_txn.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	g.Record(summary.RestoreDataSize, archiveSize)
+
+	if len(files) == 0 {
+		log.Info("all files are filtered out from the backup archive, nothing to restore")
+		return nil
+	}
+	summary.CollectInt("restore files", len(files))
+
+	ranges, _, err := restore.MergeFileRanges(
+		files, conn.DefaultMergeRegionSizeBytes, conn.DefaultMergeRegionKeyCount)
```
Comment (Leavrth) [inline br/pkg/task/restore_txn.go:76]: please add `TODO` if it supports vaults from config in the future.

**Review by Leavrth — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/restore/client.go [hunk 3]
```diff
@@ -355,23 +355,27 @@ func (rc *Client) SetStorage(ctx context.Context, backend *backuppb.StorageBacke
 	return nil
 }
 
-func (rc *Client) InitClients(backend *backuppb.StorageBackend, isRawKvMode bool) {
+func (rc *Client) InitClients(backend *backuppb.StorageBackend, isRawKvMode bool, isTxnKvMode bool) {
```
[the remark this answers is not in the document]
Comment (3pointer) [reply br/pkg/restore/client.go:358]: I think we can merge these `isRawKvMode, isTxnKvMode, rc.rewriteMode` into one `Context`. what about make it in another PR.

**Comment by hawkingrei** [comment]
@3pointer have a built error in CI

**Comment by 3pointer** [comment]
> @3pointer have a built error in CI

It's strange. I cloud run `make bazel_prepare` success in my local env.

**Comment by fancy-lee** [comment]
Hi，does br txn support incremental backup? How to use br txn for incremental backup?

**Pull request merged**